pipeline_steps/gate.groovy

```diff
@@ -0,0 +1,363 @@
+            }
+            println("Base branch being tested against: ${ghprbTargetBranch}.")
+            pullRequestIDs = antecedentTriggerBuilds.collect(){getPullRequestID(it)}
+            pullRequestIDs.add(getPullRequestID(triggerBuild))
+            println("Pull requests to merge on top of base branch in tests: ${pullRequestIDs}.")
+        }
+
+        // A failed gate build only causes the trigger build to fail if all antecedents succeed
+        if (antecedentTriggerBuilds.empty) {
+            failedGateBuilds = getFailedGateBuilds()
+            if (failedGateBuilds) {
+                errMsg = "One or more builds failed to reach success:\n${failedGateBuilds.collect() {it.getAbsoluteUrl()}.join("\n")}"
+                println(errMsg)
+                throw new REException(errMsg)
+            }
+        }
+
+        pullRequestIDsParam = common.dumpCSV(pullRequestIDs)
+        def parallelBuilds = [:]
+        for (j in gateJobs) {
+            WorkflowJob job = j
+            if (! (job in gateBuilds)) {
+               gateBuilds[job] = [
+                    "build": null,
+                    "nextNumber": job.getNextBuildNumber(),
+                ]
+                String jobName = job.displayName
+                parallelBuilds[jobName] = {
+                    build(
+                        job: jobName,
+                        wait: false,
+                        parameters: [
+                            [
+                                $class: "StringParameterValue",
+                                name: "RPC_GATING_BRANCH",
+                                value: RPC_GATING_BRANCH,
+                            ],
+                            [
+                                $class: "StringParameterValue",
+                                name: "BRANCH",
+                                value: ghprbTargetBranch,
+                            ],
+                            [
+                                $class: "StringParameterValue",
+                                name: "pullRequestChain",
+                                value: pullRequestIDsParam,
+                            ],
+                        ]
+                    )
+                }
+            }
+        }
+        if (parallelBuilds){
+            parallel parallelBuilds
+        }
+        if (antecedentTriggerBuilds.empty && isGateBuildsSuccess()){
+            break
+        }else {
+            sleep(time: 120, unit: "SECONDS")
+        }
+    }
+
+    (pr_repo_org, pr_repo_name) = ghprbGhRepository.split("/")
+    /* When a check is required by GitHub, it prevents the pull request being merged if
+       it is not marked as `"success"`. Normally GHPRB would be soley responsible for
+       updating the status context however this would necessitate a separate job to
+       perform the merge. The following section updates the pull request's status
+       context on GitHub if all tests were successful to enable the same bulid to merge
```

Comment:
Nit: build

Reply:
Don't tell me you don't know what a bulid is...

Reply:
done